src/transaction.rs

```diff
@@ -125,44 +125,28 @@ impl<'a> Transaction<'a> {
     }
 
     fn load_build_app_cache(&self, app: &crate::models::Application) -> Result<(), EngineError> {
-        // do load build cache before building app
-        let build = app.to_build();
-        let _ = match self.engine.build_platform().has_cache(&build) {
-            Ok(CacheResult::MissWithoutParentBuild) => {
-                info!("first build for app {} - cache miss", app.name.as_str());
-            }
-            Ok(CacheResult::Hit) => {
-                info!("cache hit for app {}", app.name.as_str());
-            }
-            Ok(CacheResult::Miss(parent_build)) => {
-                info!("cache miss for app {}", app.name.as_str());
-
-                let container_registry = self.engine.container_registry();
-
-                // pull image from container registry
-                // FIXME: if one day we use something else than LocalDocker to build image
-                // FIXME: we'll need to send the PullResult to the Build implementation
-                let _ = match container_registry.pull(&parent_build.image) {
-                    Ok(pull_result) => pull_result,
-                    Err(err) => {
-                        warn!(
-                            "{}",
-                            err.message.clone().unwrap_or(format!(
-                                "something goes wrong while pulling image from {:?} container registry",
-                                container_registry.kind()
-                            ))
-                        );
-                        return Err(EngineError::new_from_legacy_engine_error(err));
-                    }
-                };
-            }
+        let container_registry = self.engine.container_registry();
+        let mut image = app.to_image();
+
+        image.tag = String::from("latest");
+        // pull image from container registry
+        // FIXME: if one day we use something else than LocalDocker to build image
+        // FIXME: we'll need to send the PullResult to the Build implementation
+        let _ = match container_registry.pull(&image) {
+            Ok(pull_result) => pull_result,
             Err(err) => {
-                warn!("load build app {} cache error: {}", app.name.as_str(), err.message());
-                return Err(err);
+                warn!(
+                    "{}",
+                    err.message.clone().unwrap_or(format!(
+                        "something goes wrong while pulling image from {:?} container registry",
+                        container_registry.kind()
+                    ))
+                );
+                return Err(EngineError::new_from_legacy_engine_error(err));
             }
         };
 
-        Ok(())
+        return Ok(());
```

Comment:
OPT: no need for explicit `return` can be `Ok(())` without `;`

Reply:
done